lib/pusher/client.rb

```diff
@@ -88,6 +105,12 @@ def encrypted?
       @scheme == 'https'
     end
 
+    def cluster=(cluster)
```

Comment:
Exactly, it raises to many questions. These options only exist because of the singleton and should be deprecated with it. If the user wants to change the config he can always create a new client and pass the options in the constructor.

Until then I think that the last option is the best, both other options need to record the ordering of independent method calls.